openlibrary/plugins/importapi/code.py

```diff
@@ -229,6 +229,7 @@ def ia_import(
                 raise_non_book_marc(marc_record)
             try:
                 edition_data = read_edition(marc_record)
+                edition_data["has_marc_record"] = True
```

Comment:
@scottbarnes I'm not sure this is a good idea. It looks like if no existing edition is found, this `has_marc_record` will get written to the newly created edition record, which isn't the intent. This looks like it is meant to be a temporary variable which is stored on the import record format ... which is risky.

Reply:
That would definitely be bad. Thank you for pointing that out. Back to the drawing board. :)

Reply:
I removed this pursuant to your suggestion about creating a more general solution.